itest/litd_custom_channels_test.go

```diff
@@ -1675,3 +1678,207 @@ func testCustomChannelsBreach(_ context.Context, net *NetworkHarness,
+	t.Logf("Opening normal channel between Dave and Erin...")
+	channelOp := openChannelAndAssert(
+		t, net, dave, erin, lntest.OpenChannelParams{
+			Amt:         5_000_000,
+			SatPerVByte: 5,
+		},
+	)
+	defer closeChannelAndAssert(t, net, dave, channelOp, false)
+
+	assertChannelKnown(t.t, charlie, channelOp)
+
+	charlieTap := newTapClient(t.t, charlie)
+	daveTap := newTapClient(t.t, dave)
+	universeTap := newTapClient(t.t, zane)
+
+	// Mint an asset on Charlie and sync Dave to Charlie as the universe.
+	mintedAssets := itest.MintAssetsConfirmBatch(
+		t.t, t.lndHarness.Miner.Client, charlieTap,
+		[]*mintrpc.MintAssetRequest{
+			{
+				Asset: itestAsset,
+			},
+		},
+	)
+	cents := mintedAssets[0]
+	assetID := cents.AssetGenesis.AssetId
+	var groupKey []byte
+	if cents.AssetGroup != nil {
+		groupKey = cents.AssetGroup.TweakedGroupKey
+	}
+	fundingScriptTree := tapchannel.NewFundingScriptTree()
+	fundingScriptKey := fundingScriptTree.TaprootKey
+	fundingScriptTreeBytes := fundingScriptKey.SerializeCompressed()
+
+	t.Logf("Minted %d lightning cents, syncing universes...", cents.Amount)
+	syncUniverses(t.t, charlieTap, dave)
+	t.Logf("Universes synced between all nodes, distributing assets...")
+
+	charlieBalance := cents.Amount
+
+	fundRespCD, err := charlieTap.FundChannel(
+		ctxb, &tchrpc.FundChannelRequest{
+			AssetAmount:        charlieBalance,
+			AssetId:            assetID,
+			PeerPubkey:         daveTap.node.PubKey[:],
+			FeeRateSatPerVbyte: 5,
+			PushSat:            0,
+		},
+	)
+	require.NoError(t.t, err)
+	t.Logf("Funded channel between Charlie and Dave: %v", fundRespCD)
+
+	// Make sure the pending channel shows up in the list and has the
+	// custom records set as JSON.
+	assertPendingChannels(
+		t.t, charlieTap.node, assetID, 1, charlieBalance, 0,
+	)
+
+	// Let's confirm the channel.
+	mineBlocks(t, net, 6, 1)
+
+	assertAssetBalance(t.t, charlieTap, assetID, cents.Amount)
+
+	// There should only be a single asset piece for Charlie, the one in the
+	// channel.
+	assertNumAssetOutputs(t.t, charlieTap, assetID, 1)
+	assertAssetExists(
+		t.t, charlieTap, assetID, charlieBalance,
+		fundingScriptKey, false, true, true,
+	)
+
+	// Assert that the proofs for both channels has been uploaded to the
+	// designated Universe server.
+	assertUniverseProofExists(
+		t.t, universeTap, assetID, groupKey, fundingScriptTreeBytes,
+		fmt.Sprintf("%v:%v", fundRespCD.Txid, fundRespCD.OutputIndex),
+	)
+
+	// Make sure the channel shows the correct asset information.
+	assertAssetChan(
+		t.t, charlieTap.node, daveTap.node, charlieBalance, assetID,
+	)
+
+	logBalance(t.t, nodes, assetID, "initial")
+
+	// Normal case.
+	// Send 50 assets from Charlie to Dave.
+	sendAssetKeySendPayment(
+		t.t, charlie, dave, 50, assetID,
+		fn.None[int64](), lnrpc.Payment_SUCCEEDED,
+	)
+
+	logBalance(t.t, nodes, assetID, "after 50 assets")
+
+	// Normal case.
+	// Send 1k sats from Charlie to Dave.
+	sendKeySendPayment(t.t, charlie, dave, 1000)
```

Comment:
Shouldn't there now be 1500 sats on Dave's side? 500 from the asset keysend above (which is hard coded for asset keysends) and the 1k here? And the reserve AFAIK is arount 1060, so there should actually be enough balance for a 354 sat HTLC?

Reply:
keysends are using 500 sats, so there's actually not enough! I was following the steps of the bug report [here](https://github.com/lightninglabs/taproot-assets/issues/1060)

Reply:
Ah, right. Might add that as a comment.